### DISCUSSION TITLE
Why does Rider behave differently from Visual Studio on the Method FindControl?

### DISCUSSION THREAD

**Discussion by LZD121** [AvaloniaUI/Avalonia#10596] [Q&A] [answered]
I followed the guide of Avalonia's Documentation which is using "FindControl" method to acquire a control from View and save it into a property.
I have a control named "TextBlockNumber1" and a property also named "TextBlockNumber1" back in my view.cs
![image](https://user-images.githubusercontent.com/59716053/223673654-4aefecdc-51e1-49b4-8bb4-2c9286b4bbdd.png)

![image](https://user-images.githubusercontent.com/59716053/223673688-10b023ed-16a5-41cb-89aa-125f1c3c2b63.png)
It works in Visual Studio. But in Ride it alerts me an error that there is already a member with the same name.
![image](https://user-images.githubusercontent.com/59716053/223674297-d079a92e-7892-47f2-964d-8fd9f6446ffe.png)

Of course with this error the program can not pass compiling...

Why does Rider stop me from doing this when I follow the Avalonia recommanded way?

**Comment by timunie** [comment]
hmmm that is really strange. For me it must be the same on either IDE. Can you prepare a minimal sample for us to double-check what's wrong?

**Reply by LZD121** [reply]
I don't think it's necessary a particular sample for this. I just created a simple test view like this:
![image](https://user-images.githubusercontent.com/59716053/223706177-8fb9e03b-3ee0-4bc6-af18-2e34c98cc1ef.png)

![image](https://user-images.githubusercontent.com/59716053/223706350-52070139-3760-45f0-848c-6b91ac6bc94f.png)

As you can see, the error is still there.

**Reply by timunie** [reply]
looks like you have the NamesGernator installed. In that case Rider is correct. But VisualStudio should behave the same. At least it should not build. You should be able to use TestButton without the need to define it again.

**Reply by LZD121** [reply]
Thanks, you are right.Indeed the VS cannot build with this. And I did installed "XamlNameReferenceGenerator"
But in VS I can not reference this control directly while in Rider I can.
Wired again...
![image](https://user-images.githubusercontent.com/59716053/223710637-70f12402-fe67-466d-aef0-cd62751b0cf0.png)

![image](https://user-images.githubusercontent.com/59716053/223710695-d6928d1f-fae0-47d5-ba60-7398bc43c08d.png)

**Reply by timunie** [reply]
yes, that source generator stuff is sometimes limited. you may need to clean the cache and rebuild the solution. I have the same issue with the CommunityMVVMToolkit. While the text is red, it compiles anyway.

**Comment by timunie** [comment, marked as answer]
Answered here https://github.com/AvaloniaUI/Avalonia/discussions/10596#discussioncomment-5240912
